runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/graph/GreedyPipelineFuserTest.java

```diff
@@ -452,26 +466,59 @@ public void flattenWithHeterogenousInputsAndOutputsEntirelyMaterialized() {
     FusedPipeline fused =
         GreedyPipelineFuser.fuse(Pipeline.newBuilder().setComponents(components).build());
 
+    assertThat(fused.getRunnerExecutedTransforms(), hasSize(3));
     assertThat(
+        "The runner should include the impulses for both languages, plus an introduced flatten",
         fused.getRunnerExecutedTransforms(),
-        containsInAnyOrder(
+        hasItems(
             PipelineNode.pTransform("pyImpulse", components.getTransformsOrThrow("pyImpulse")),
             PipelineNode.pTransform("goImpulse", components.getTransformsOrThrow("goImpulse"))));
+
+    PTransformNode flattenNode = null;
+    for (PTransformNode runnerTransform : fused.getRunnerExecutedTransforms()) {
+      if (!(runnerTransform.getId().equals("pyImpulse")
+          || runnerTransform.getId().equals("goImpulse"))) {
+        flattenNode = runnerTransform;
```

Comment:
Yes; we still need the loop, because the fuser doesn't care what the id of the introduced flatten is, but we can find the is the flatten of the partial PCollections via inspecting its output.